go/ros/bag2mcap.go

```diff
@@ -254,14 +276,15 @@ func Bag2MCAP(w io.Writer, r io.Reader, opts *mcap.WriterOptions) error {
 		return err
 	}
 	seq := uint32(0)
+	connIds := make([]uint32, 0)
 	schemas := make(map[string]uint16)
 	return processBag(r,
 		func(header, data []byte) error {
 			conn, err := extractHeaderValue(header, headerConn)
 			if err != nil {
 				return err
 			}
-			connID := binary.LittleEndian.Uint16(conn)
+			connID := binary.LittleEndian.Uint32(conn)
```

Comment:
instead of adding the scan to each message iteration, what about just doing something like this?

```
func safeUint32Downcast(x uint32) (uint16, error) {
  if x > math.MaxUint16 {
    return 0, ErrTooManyConnections
  }
  return uint16(x), nil
}
```

then call it right here -

```
chanID, err := safeUint32Downcast(binary.LittleEndian.Uint32(conn)))
if err != nil {
  return err
}
...
```

this gets us constant time performance in connections - no need to traverse the list of connIds each time. The only thing it hurts us on afaict is if someone is assigning connection IDs out of sequence, _and_ surpassing this limit. I think that seems unlikely enough that we can wait for the first user report to attempt to accommodate it. The change in behavior to crash when this happens instead of silently corrupting data, is already a huge improvement IMO.